### DISCUSSION TITLE
how would I refactor a build script that uses &&

### DISCUSSION THREAD

**Discussion by dagda1** [nrwl/nx#7763] [General]
I have the following build script

```json
    "generate": "nx run-many --target=generate --all --parallel",
    "tsc:build": "tsc --build",
    "build": "yarn generate && yarn tsc:build && nx run-many --target=build --all --parallel",
  ```

Will using `&&` make the build slower and not take advantage of caching?  

If so, I would appreciate some advice on how to refactor this.

**Comment by eppsilon** [comment]
Your build/generate targets will still be parallelized. The build ones are cached by default. If you want the generate targets to be cached too, add them to `cacheableOperations` in nx.json.

You could add a nx project/target for your `tsc:build` command so it can be cached as well. There's [an executor for running `tsc` specifically](https://nx.dev/l/a/workspace/tsc) and one for [running custom commands](https://nx.dev/l/a/workspace/run-commands-executor).

**Comment by dagda1** [comment]
@eppsilon I see this line a lot in the docs 

> Options can be configured in angular.json when defining the executor

I am not working with angular so this is not very useful.  

I can see how I add an executor to a project but what about adding an executor at the root of a monorepo?  How is that done?

Are there schemas for nx.json and workspace.json that I could add to vscode?

**Comment by eppsilon** [comment]
Non-Angular NX repos use [workspace.json](https://nx.dev/l/r/core-concepts/configuration#workspacejson-and-projectjson) instead of angular.json. They serve the same purpose, though.

I don't think it is possible to add an executor without a containing project.
